moveit_ros/occupancy_map_monitor/src/occupancy_map_monitor.cpp

```diff
@@ -39,124 +39,90 @@
 #include <moveit_msgs/srv/load_map.hpp>
 #include <moveit/occupancy_map_monitor/occupancy_map.h>
 #include <moveit/occupancy_map_monitor/occupancy_map_monitor.h>
+#include <moveit/occupancy_map_monitor/occupancy_map_monitor_middleware_handle.hpp>
 #include <boost/bind.hpp>
 
 namespace occupancy_map_monitor
 {
+namespace
+{
 static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit.ros.occupancy_map_monitor");
+}
 
 OccupancyMapMonitor::OccupancyMapMonitor(const rclcpp::Node::SharedPtr& node, double map_resolution)
-  : map_resolution_(map_resolution), debug_info_(false), mesh_handle_count_(0), node_(node), active_(false)
+  : OccupancyMapMonitor(std::make_unique<OccupancyMapMonitorMiddlewareHandle>(node), nullptr, "", map_resolution)
 {
-  initialize();
 }
 
 OccupancyMapMonitor::OccupancyMapMonitor(const rclcpp::Node::SharedPtr& node,
                                          const std::shared_ptr<tf2_ros::Buffer>& tf_buffer,
                                          const std::string& map_frame, double map_resolution)
-  : tf_buffer_(tf_buffer)
+  : OccupancyMapMonitor(std::make_unique<OccupancyMapMonitorMiddlewareHandle>(node), tf_buffer, map_frame,
+                        map_resolution)
+{
+}
+
+OccupancyMapMonitor::OccupancyMapMonitor(std::unique_ptr<MiddlewareHandle> middleware_handle,
+                                         const std::shared_ptr<tf2_ros::Buffer>& tf_buffer,
+                                         const std::string& map_frame, double map_resolution)
+  : middleware_handle_(std::move(middleware_handle))
+  , tf_buffer_(tf_buffer)
   , map_frame_(map_frame)
   , map_resolution_(map_resolution)
   , debug_info_(false)
   , mesh_handle_count_(0)
-  , node_(node)
-{
-  initialize();
-}
-
-void OccupancyMapMonitor::initialize()
+  , active_(false)
 {
-  /* load params from param server */
-  if (map_resolution_ <= std::numeric_limits<double>::epsilon())
-    if (!node_->get_parameter("octomap_resolution", map_resolution_))
-    {
-      map_resolution_ = 0.1;
-      RCLCPP_WARN(LOGGER, "Resolution not specified for Octomap. Assuming resolution = %g instead", map_resolution_);
-    }
+  map_resolution_ = middleware_handle_->getMapResolutionParameter();
```

Comment:
It looks like the original version checks the ctor set value first, whereas this version overrides it regardless. Was this the intent? If so, what's the point of the ctor argument?

Reply:
I didn't mean to change the behavior, putting the old  behavior back.